x-pack/plugins/enterprise_search/public/applications/app_search/components/app_search_gate/app_search_gate.tsx

```diff
@@ -0,0 +1,616 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+
+import { useActions, useValues } from 'kea';
+
+import {
+  EuiButton,
+  EuiCallOut,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiForm,
+  EuiFormLabel,
+  EuiFormRow,
+  EuiIcon,
+  EuiLink,
+  EuiPanel,
+  EuiSelect,
+  EuiSpacer,
+  EuiSuperSelect,
+  EuiText,
+  EuiTextArea,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+
+import { FormattedMessage } from '@kbn/i18n-react';
+
+import { docLinks } from '../../../shared/doc_links';
+
+import { AppSearchGateLogic } from './app_search_gate_logic';
+
+const featuresList = {
```

Comment:
Non blocking suggestion.  Not sure if already discussed, Workplace search & app search seems to use same form only the options & wordings are a bit different. In this case, May be we can move this file and may be the [app search gate page](https://github.com/elastic/kibana/blob/bce8bcfeb3ac0ee9490d4854da44204b68a9d633/x-pack/plugins/enterprise_search/public/applications/app_search/components/app_search_gate/app_search_gated_form_page.tsx)  to [shared](https://github.com/elastic/kibana/tree/bce8bcfeb3ac0ee9490d4854da44204b68a9d633/x-pack/plugins/enterprise_search/public/applications/shared) folder ? So that workplace search & app search use can use  the`gated_form` component and pass options as props which would differentiate the solution. 

I think this would make the component reusable, easy to maintain and make it more clean. WDYT?

Reply:
I like the idea - however, we won't need to add any more of these forms in the future... and this will only be in the 8.x versions... do you foresee something where the maintainability of this may be affected by not having this?

Reply:
> do you foresee something where the maintainability of this may be affected by not having this?

No I don't, but moving to shared would help if we want to make changes to both workplace search & app search gate form.